### PR TITLE
HLA-1269 Testing with Python 3.12

### LINKED ISSUE #1673: TST: Test against numpy 2.0.dev and Python 3.12
Should do it sooner than later. Probably should prioritize Python 3.12 if you cannot do both at the same time.

### PR DESCRIPTION
**Pull request by s-goldman**
Resolves [HLA-1269](https://jira.stsci.edu/browse/HLA-1269)

Closes #1673

This PR includes new testing for python 3.12 for Jenkins and github actions. Several other tests were added for github actions and both the CI and CI_CRON github actions workflow files were rearrange for consistency and the JenkinsRT file was reorganized for simplicity. 

**Checklist for maintainers**
- [x] added entry in `CHANGELOG.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)

[Jenkins test](https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FDrizzlepac-Developers-Pull-Requests/detail/Drizzlepac-Developers-Pull-Requests/111/pipeline/339/)

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -20,6 +20,9 @@ number of the code change for that issue.  These PRs can be viewed at:
 
 3.7.1 (unreleased)
 ==================
+
+- Added python 3.12 to testing matrix for Jenkins and github actions. [#1843]
```
Comment (mdlpstsci) [inline CHANGELOG.rst:24]: Please move the python 3.12 comment out of section 3.7.1 and put it in a new section 3.7.2.

**Review by mdlpstsci — DISMISSED** [review]
Aside from the small text placement in the CHANGELOG.rst (**_Thank you_**), these changes look fine to me.  However, since I do not usually handle this infrastructure stuff, I think it is better if Zach or Joe reviews this ticket.

**Comment by s-goldman** [comment]
> Aside from the small text placement in the CHANGELOG.rst (**_Thank you_**), these changes look fine to me. However, since I do not usually handle this infrastructure stuff, I think it is better if Zach or Joe reviews this ticket.

@zacharyburnett @jhunkeler

File: CHANGELOG.rst [hunk 2]
```diff
@@ -18,13 +18,20 @@ number of the code change for that issue.  These PRs can be viewed at:
...
 ==================
-- Reverted #1798 until further testing is done with Photutils.
+
+- Added python 3.12 to testing matrix for Jenkins and github actions. [#1843]
 
 - ``manageInputCopies`` now copies successfully even if the original files were
   defined by full paths rather than being in the current working directory. [#1835]
 
+
+3.7.1 (unreleased)
```
Comment (jhunkeler) [inline CHANGELOG.rst:30]: Should 3.7.1 (unreleased anyway) be rolled up into 3.7.2? Or should 3.7.2 above be changed to 3.7.1.

File: JenkinsfileRT [hunk 3]
```diff
@@ -25,56 +25,50 @@ data_config.server_id = 'bytesalad'
...
+// Test 1 ///////////////////////////
 bc1 = new BuildConfig()
 bc1.nodetype = 'linux'
 bc1.env_vars = ['TEST_BIGDATA=https://bytesalad.stsci.edu/artifactory']
 bc1.name = '3.10'
 bc1.conda_packages = ['python=3.10']
 bc1.build_cmds = ["pip install numpy astropy pytest-cov ci-watson",
-		 "pip install --upgrade -e '.[test]'",
-                 "pip freeze"]
+                "pip install --upgrade -e '.[test]'",
```
Comment (jhunkeler) [inline JenkinsfileRT:35]: Can we excise this `\t` character? ... Oh the PR merged while I was writing this.

**Review by jhunkeler — APPROVED** [review]

**Pull request merged**